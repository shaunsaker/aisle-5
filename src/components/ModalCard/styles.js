import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    backgroundColor: styleConstants.colors.white,
    borderRadius: styleConstants.dimensions.borderRadius,
    marginVertical: styleConstants.dimensions.spacing.vertical,
    marginHorizontal: styleConstants.dimensions.spacing.horizontal,
    paddingVertical: styleConstants.dimensions.spacing.vertical * 2,
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal * 2,
    ...styleConstants.shadows.large,
    maxWidth: 480, // tablets
  },
});

export default styles;
