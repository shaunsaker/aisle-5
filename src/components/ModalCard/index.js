import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import styles from './styles';

const propTypes = {
  children: PropTypes.node,
};

const defaultProps = {};

const ModalCard = ({ children }) => {
  return <View style={styles.container}>{children}</View>;
};

ModalCard.propTypes = propTypes;
ModalCard.defaultProps = defaultProps;

export default ModalCard;
