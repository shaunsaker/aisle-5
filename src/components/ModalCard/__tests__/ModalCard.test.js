import React from 'react';
import renderer from 'react-test-renderer';

import ModalCard from '..';

describe('ModalCard', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<ModalCard />);

      expect(component).toMatchSnapshot();
    });
  });
});
