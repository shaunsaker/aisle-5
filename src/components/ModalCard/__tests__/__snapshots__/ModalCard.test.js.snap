// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalCard renders renders with minimum required props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EDE8CA",
      "borderColor": null,
      "borderRadius": 5,
      "borderWidth": 0,
      "elevation": 12,
      "marginHorizontal": 15,
      "marginVertical": 10,
      "maxWidth": 480,
      "paddingHorizontal": 30,
      "paddingVertical": 20,
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.33,
      "shadowRadius": 4,
    }
  }
/>
`;
